File: tests/expectations/ptrs_as_arrays.java
import com.sun.jna.*;
import com.sun.jna.ptr.*;

enum BindingsSingleton {
  INSTANCE;
  final Bindings lib = Native.load("", Bindings.class);
}

interface Bindings extends Library {
  Bindings INSTANCE = BindingsSingleton.INSTANCE.lib;

  void ptr_as_array(int n, IntByReference arg, LongByReference v);

  void ptr_as_array1(int n, IntByReference arg, LongByReference v);

  void ptr_as_array2(int n, IntByReference arg, LongByReference v);

  void ptr_as_array_wrong_syntax(IntByReference arg, IntByReference v, IntByReference arg2);

  void ptr_as_array_unnamed(IntByReference arg0, IntByReference arg1);

}

File: tests/expectations/ptrs_as_arrays.zig
const std = @import("std");

extern fn ptr_as_array(n: u32, arg: [3]u32, _v: ?*u64) anyopaque;

extern fn ptr_as_array1(n: u32, arg: [3]u32, v: [4]u64) anyopaque;

extern fn ptr_as_array2(n: u32, arg: [*]u32, v: [*]u64) anyopaque;

extern fn ptr_as_array_wrong_syntax(_arg: ?*u32, _v: ?*u32, _: ?*u32) anyopaque;

extern fn ptr_as_array_unnamed(_: ?*u32, _: ?*u32) anyopaque;
